fix(autoware_simple_object_merge): templatize simple object merger

a-maumau opened the pull request:
## Description

## Related links

**Parent Issue:**

- Link

<!-- ⬇️🟢
**Private Links:**

- [CompanyName internal link]()
⬆️🟢 -->

## How was this PR tested?

## Notes for reviewers

None.

## Interface changes

None.

<!-- ⬇️🔴

### Topic changes

#### Additions and removals

| Change type   | Topic Type      | Topic Name    | Message Type        | Description       |
|:--------------|:----------------|:--------------|:--------------------|:------------------|
| Added/Removed | Pub/Sub/Srv/Cli | `/topic_name` | `std_msgs/String`   | Topic description |

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub/Srv/Cli | `/old_topic_name` | `sensor_msgs/Image` | Topic description |
| New     | Pub/Sub/Srv/Cli | `/new_topic_name` | `sensor_msgs/Image` | Topic description |

### ROS Parameter Changes

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added/Removed | `param_name`   | `double` | `1.0`         | Param description |

#### Modifications

| Version | Parameter Name   | Type     | Default Value | Description       |
|:--------|:-----------------|:---------|:--------------|:------------------|
| Old     | `old_param_name` | `double` | `1.0`         | Param description |
| New     | `new_param_name` | `double` | `1.0`         | Param description |

🔴⬆️ -->

## Effects on system behavior

None.

technolojin reviewed (changes requested):
1. Considering the original is used broadly, I would like to request not to change the node name.

detected_object_simple_merger -> simple_object_merger




2. 'simple' is prefix to indicate the algorithm. therefore,

tracked_object_simple_merger -> simple_tracked_object_merger



4. need to pair with update of https://github.com/tier4/aip_launcher/blob/e990dd10095f17ce5edd83b92f4f1e45986da431/aip_x2_gen2_launch/launch/radar.launch.xml#L196-L201

a-maumau commented:
@technolojin 
Thank you for your suggestions.
I fixed the names so that "simple" comes first, and reverted the node name for detected objects in https://github.com/autowarefoundation/autoware_universe/pull/10785/commits/3a0b3335d17c12ab920b3d90ab0b00df3fcc4d53.
Since the original `simple_object_merger` node remains, we don't need to update `aip_launcher` for now.

technolojin reviewed (approved):
LGTM
